security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -322,12 +325,16 @@ func (s *sdsservice) StreamSecrets(stream sds.SecretDiscoveryService_StreamSecre
 				"error details %s\n", conIDresourceNamePrefix, discReq.Node.Id, firstRequestFlag, discReq.VersionInfo,
 				discReq.ErrorDetail)
 
-			// In ingress gateway agent mode, if the first SDS request is received but kubernetes secret is not ready,
-			// wait for secret before sending SDS response. If a kubernetes secret was deleted by operator, wait
-			// for a new kubernetes secret before sending SDS response.
-			if s.st.ShouldWaitForIngressGatewaySecret(conID, resourceName, token) {
+			// In ingress gateway agent mode, if the first SDS request is received but Ingress gateway secret which is
+			// setup as kubernetes secret is not ready, wait for secret before sending SDS response.
```

Comment:
@lei-tang In this comment
"// In ingress gateway agent mode, if the first SDS request is received but Ingress gateway secret which is
// setup as kubernetes secret is not ready, wait for secret before sending SDS response."

I explained how Ingress gateway secret is setup i.e. as kubernetes secret
In the below I explained how file mounted certs are setup with the env flag.

If you are looking for any this else to be documented - please suggest. I will add

Reply:
With the existing ingress gateway secret mechanism, why the mechanism of file mounted certs is needed for ingress gateway?

Reply:
I just added some more explanation. PTAL. Please suggest if if you have other points. Will add

Reply:
I am not sure which explanation in your PR answers the question: with the existing ingress gateway secret mechanism, why the mechanism of file mounted certs is needed for ingress gateway?

To clearly anser the question, please paste your answer to the above question as a comment.